### ISSUE TITLE
Can not select drawing widgets

### ISSUE THREAD

**Issue by billreed63** [maqetta/maqetta#2426]
Copy and paste below html file:

```
<!DOCTYPE html>
<html>
<head>
<meta charset="utf-8"/>
<title>Untitled</title>
<script type="text/javascript" src="lib/dojo/dojo/dojo.js" data-dojo-config="'async':true,'parseOnLoad':true,'packages':[{'name':'maqetta','location':'../../maqetta'},{'name':'clipart','location':'../../clipart'},{'name':'shapes','location':'../../shapes'},{'name':'jquery-ui','location':'../../jquery-ui'},{'name':'yui','location':'../../yui'}]"></script>
<script type="text/javascript">
require([
  "dijit/dijit",
  "dojo/parser",
  "maqetta/space",
  "maqetta/AppStates",
  "shapes/Rectangle",
  "shapes/Text",
  "shapes/Line"
]);
</script>
<style>@import "themes/claro/document.css";@import "themes/claro/claro.css";@import "lib/shapes/shapes.css";@import "app.css";
</style>
</head>
<body class="claro" dvFlowLayout="false" data-davinci-ws="collapse" id="myapp">
<span data-dojo-type="shapes.Rectangle" width="693" height="376" style="position: absolute; z-index: 900; width: 695px; height: 378px; left: 81px; top: 28px;"></span><span data-dojo-type="shapes.Rectangle" width="119" height="31" cornerRadius="10" style="position: absolute; z-index: 900; left: 633px; top: 348px; width: 121px; height: 33px;"></span><span data-dojo-type="shapes.Rectangle" width="119" height="31" cornerRadius="10" style="width: 121px; height: 33px; position: absolute; z-index: 900; left: 489px; top: 348px;"></span><span data-dojo-type="shapes.Text" style="font-size: 14px; position: absolute; z-index: 900; left: 510px; top: 354.5px;" content="Ok"></span><span data-dojo-type="shapes.Text" content="Cancel" style="font-size: 14px; position: absolute; z-index: 900; left: 659px; top: 354.5px;"></span><span data-dojo-type="shapes.Text" style="font-size: 14px; position: absolute; z-index: 900; left: 116px; top: 50px;" content="My stuff"></span><span data-dojo-type="shapes.Line" startarrow="none" style="position: absolute; z-index: 900; left: 116px; top: 88px; width: 645px; height: 2px;" points="0,3,643,0"></span></body>
</html>

```

Refresh browser

Select on of the "button" looking rectangles then select the rectangle that looks like a dialog, now I can no longer select any of the other widgets.. I can select the widget from the outline but not the page editor

**Comment by JonFerraiolo** [comment]
If you unselect the bigger rectangle (the one that looks like a dialog), you can then select the inner "buttons". The outer selection is somehow capturing the mousedown event before the inner shapes get the mousedown event. I think it's because the bigger rectangle is a primitive shape (even though it looks like a container), and the selection logic on primitive widgets is that the widget consumes all mouse events (in preparation for a possible drag operation) when that primitive widget is selected.

Maybe this elevates the priority of #2051, which would take care of this problem.

**Comment by JonFerraiolo** [comment]
I found the culprit. It's these lines in SelectTool.js:

```
        var widget = context.checkFocusXY(event.pageX, event.pageY);
        if(widget && Metadata.getAllowedChild(widget.type)[0] !== 'NONE'){
            widget = null;
        }
```

The above lines deal with some edge cases where the user has down a mousedown within the widget's layout margin bounds (using CSS box model) but because of particular set of circumstances the event.target isn't a node within the widget's dom tree (e.g., if you are clicking in the margin of the widget or if CSS transforms have moved the widget). If we remove the above lines, then this bug goes away, but other bugs crop up again.

We could special-case the shapes widgets somehow, but my current thinking is to fix #2051, which is too big of a change for any sort of Preview 6 dot release.

**Comment by JonFerraiolo** [comment]
Fixed by 128379cadeb6428ac665f4ef6c0b4ffa0bf896db
